homework/index.js

```diff
@@ -30,17 +35,186 @@
+    createAndAppend('h4', widgetContainer, {
+      text: `${options.title}:`,
+      class: 'py-2 px-2 widget-title',
+    });
+    createAndAppend(options.valueTag || 'h4', widgetContainer, {
+      text: options.value,
+      class: 'widget-value lead',
+    });
+    return widgetContainer;
+  }
+
+  function fetchContributor(url) {
+    fetchJSON(url)
+      .then(contributorsData => {
+        createContributorsLayout(contributorsData);
+      })
+      .catch(err => renderError(err));
+  }
+
+  function createRepositoryLayout(repositoriesData) {
+    const mainParent = document.getElementById('main');
+    const leftColumn = createAndAppend('div', mainParent, {
+      class: 'd-flex flex-column repo-detail',
+      id: 'repo-detail',
+    });
+
+    const widgetContainer = createRepositoryWidget(leftColumn, {
+      faClasses: 'fab fa-github',
+      title: 'Repository Name:',
+      value: repositoriesData.name,
+      valueTag: 'a',
+    });
+
+    const anchorTag = widgetContainer.lastChild;
+    anchorTag.setAttribute('href', repositoriesData.html_url);
+    anchorTag.setAttribute('target', '_blank');
+
+    createRepositoryWidget(leftColumn, {
+      faClasses: 'fas fa-pen-alt',
+      title: 'Description',
+      value: repositoriesData.description,
+    });
+
+    const createdDate = new Date(repositoriesData.created_at).toDateString();
+    createRepositoryWidget(leftColumn, {
+      faClasses: 'fas fa-calendar-plus',
+      title: 'Created',
+      value: createdDate,
+    });
+
+    const updatedDate = new Date(repositoriesData.updated_at).toDateString();
+    createRepositoryWidget(leftColumn, {
+      faClasses: 'fas fa-clock',
+      title: 'Updated',
+      value: updatedDate,
+    });
+
+    createRepositoryWidget(leftColumn, {
+      faClasses: 'fas fa-code-branch',
+      title: 'Forks',
+      value: repositoriesData.forks,
+    });
+  }
+
+  function appendRepositoriesToSelect(repositoriesData) {
+    const selectList = document.getElementById('repo-select');
+    repositoriesData
+      .sort((a, b) => a.name.localeCompare(b.name))
+      .forEach((repository, index) => {
+        createAndAppend('option', selectList, {
+          text: repository.name,
+          value: index,
+        });
+      });
+
+    createRepositoryLayout(repositoriesData[0]);
+    fetchContributor(repositoriesData[0].contributors_url);
+
+    selectList.addEventListener('change', () => {
+      selectList.parentElement.nextElementSibling.innerHTML = '';
+      createRepositoryLayout(repositoriesData[selectList.value]);
+      fetchContributor(repositoriesData[selectList.value].contributors_url);
+    });
+  }
+
+  function fetchRepository(url) {
+    fetchJSON(url)
+      .then(repositoryData => {
+        appendRepositoriesToSelect(repositoryData);
+      })
+      .catch(err => renderError(err));
+  }
+
+  function createHeader() {
+    const header = createAndAppend('header', root, {
+      class: 'container jumbotron mt-3 mb-3 px-5',
```

Comment:
Nice layout of your code.  Defining and storing in a constant and then appending.  Also makes debugging much easier.  Nice work.
And the code is very neatly laid out.